### PR TITLE
Silly Player Names Rewrite

### PR DESCRIPTION
**Pull request by theorangeangle**

### CODE REVIEW

**Comment by Zahir-Khan** [comment]
Don't have permissions to edit...
line 53: local function create_name(words_table)
line 66: local function create_unique_name(words_table)
line 68: local name = create_name(words_table)
line 71: name = create_name(words_table)
line 96: local name = create_unique_name(naming_words)
line 130: local player = game.get_player(event.player_index)

**Comment by Zahir-Khan** [comment]
line 58: local name = format('%s%s%s', adverb, adjective, noun)
line 59: return name

File: map_gen/shared/silly_player_names.lua [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+local ScenarioInfo = require 'features.gui.info'
+local Command = require 'utils.command'
+local Ranks = require 'resources.ranks'
+
+local format = string.format
+local random = math.random
+
+ScenarioInfo.add_map_extra_info('- On this map you will be assigned a silly name.\n' .. '- If you dislike your name you can /name-restore or /name-roll for a new one')
+
+global.silly_regulars = {}
```
Comment (mheguy) [inline map_gen/shared/silly_player_names.lua:17]: This doesn't do anything. I wrote this over a year ago for a very different user_group system.

File: map_gen/shared/silly_player_names.lua [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        data_silly_names = data_silly_names
+    },
+    function(tbl)
+        data_silly_names = tbl.data_silly_names
+    end
+)
+
+--- Takes a player's real name, current silly name, and old silly name and adjusts
+-- the silly_regulars table accordingly
+local function check_regular(real_name, silly_name, old_silly_name)
```
Comment (mheguy) [inline map_gen/shared/silly_player_names.lua:39]: This entire function does nothing with the current user system.

File: map_gen/shared/silly_player_names.lua [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        global.silly_regulars[silly_name] = true
+        if old_silly_name then
+            global.silly_regulars[old_silly_name] = nil
+        end
+    end
+end
+
+--- Creates name by combining elements from the passed table
+-- @param words_table including adverbs, adjectives, and nouns
+-- @param player_name string with player's name
```
Comment (mheguy) [inline map_gen/shared/silly_player_names.lua:50]: Outdated doc

File: map_gen/shared/silly_player_names.lua [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    adverb = words_table.adverbs[random(#words_table.adverbs)]
+    adjective = words_table.adjectives[random(#words_table.adjectives)]
+    noun = words_table.nouns[random(#words_table.nouns)]
+    local name = format('%s%s%s', adverb, adjective, noun)
+    return name
+end
+
+--- Calls create_name until a unique name is returned
+-- @param words_table including adverbs, adjectives, and nouns
+-- @param player_name string with player's name
```
Comment (mheguy) [inline map_gen/shared/silly_player_names.lua:64]: Outdated doc

File: map_gen/shared/silly_player_names.lua [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    data_silly_names.silly_names[#data_silly_names.silly_names + 1] = name
+    data_silly_names.silly_names_count[1] = data_silly_names.silly_names_count[1] + 1
+
+    local str = format('%s will now be known as: %s', player.name, name)
+    game.print(str)
+    local admin_str = format('%s (ID: %s)', str, player.index)
+    Utils.print_admins(admin_str, nil)
+    player.name = name
+
+    -- After they have their name, we need to ensure compatibility with the regulars system
```
Comment (mheguy) [inline map_gen/shared/silly_player_names.lua:106]: Outdated as the regulars/permission system has changed.

**Review by mheguy — CHANGES_REQUESTED** [review]
Silly player names had a fatal flaw, and I have no idea if changes to factorio have resolved that issue. When players connect to a server, the server checks if the connecting client's `name` matches a current LuaPlayer's name in game.players. If there is no corresponding name, a new LuaPlayer is created for the connecting client.
So let's say we have `orange` who connects and has their name changed to `not_orange`. The server "is shut down for maintenance" (read: crashes) while `orange` is online. When the server comes back online and `orange` reconnects, the server does not have a `LuaPlayer` named `orange`, therefore a new LuaPlayer will be connected and `orange` will not have any of the inventory that `not_orange` has. Further, they will not receive the name `not_orange` again unless by random chance.

So assuming Wube hasn't changed how players are identified to servers: this module should come with a significant warning to let people know that it is not stable.

File: map_gen/shared/silly_player_names.lua [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+    local name = create_unique_name(naming_words)
+    data_silly_names.silly_names[#data_silly_names.silly_names + 1] = name
+    data_silly_names.silly_names_count[1] = data_silly_names.silly_names_count[1] + 1
+
+    local str = format('%s will now be known as: %s', player.name, name)
+    game.print(str)
+    local admin_str = format('%s (ID: %s)', str, player.index)
+    Utils.print_admins(admin_str, nil)
+    player.name = name
```
Comment (SimonFlapse) [inline map_gen/shared/silly_player_names.lua:104]: From Factorio version 0.18.7 we are no longer able to change a players name. https://lua-api.factorio.com/latest/LuaPlayer.html#LuaPlayer.name

An alternative would be to render (Using the luaRendering) the silly names above the players head. How to handle the chat messages I don't know at the moment.

File: map_gen/shared/silly_player_names.lua [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        local str = silly_name .. ' will now be known as: ' .. player.name
+        Utils.print_admins(str .. ' (ID: ' .. player.index .. ')', nil)
+    end
+end
+
+--- Passes _event_ on to name_players
+local function player_joined(event)
+    local player = game.get_player(event.player_index)
+    if data_silly_names.silly_name_store[event.player_index] then
+        player.name = data_silly_names.silly_name_store[event.player_index]
```
Comment (SimonFlapse) [inline map_gen/shared/silly_player_names.lua:132]: API no longer available, see previous comment

File: map_gen/shared/silly_player_names.lua [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+local function restore_name(data, command_player)
+    local player
+    if data and data.player_index then
+        player = Game.get_player_by_index(data.player_index)
+    else
+        player = command_player
+    end
+    local silly_name = player.name
+    data_silly_names.silly_name_store[player.index] = player.name
+    player.name = data_silly_names.actual_name[player.index]
```
Comment (SimonFlapse) [inline map_gen/shared/silly_player_names.lua:120]: API no longer available, see previous comment

**Review by SimonFlapse — CHANGES_REQUESTED** [review]

**Comment by theorangeangle** [comment]
Seeing as how we can no longer change names, i'm going to move this one to the graveyard - it's going to take more effort than it's worth at this point to bring back into a usable state.

**Pull request closed without merging**